plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/SourceRole.kt

```diff
@@ -11,9 +11,20 @@ import org.jetbrains.kotlin.fir.types.ConeKotlinType
 import org.jetbrains.kotlin.formver.viper.ast.Info
 import org.jetbrains.kotlin.formver.viper.ast.unwrap
 import org.jetbrains.kotlin.formver.viper.errors.ErrorReason
+import org.jetbrains.kotlin.formver.viper.errors.VerificationError
 import org.jetbrains.kotlin.formver.viper.errors.extractInfoFromFunctionArgument
+import org.jetbrains.kotlin.formver.viper.errors.extractInfoFromMethodArgument
 
 sealed interface SourceRole {
+    data object ListElementAccessCheck : SourceRole {
+        /**
+         * Given a verification error on a list function, fetch the list that may cause
+         * an index out of bound error.
+         */
+        fun VerificationError.fetchOutOfBoundsIndexList(): FirBasedSymbol<*> =
+            extractInfoFromMethodArgument(0).unwrap<FirSymbolHolder>().firSymbol
```

Comment:
Is this always going to be a `FirSymbolHolder`?  What if the argument expression is something else, e.g. a function call that returns a list?

Reply:
That's true, I should check this case with some tests!